lib/crypto.ts

```diff
@@ -0,0 +1,32 @@
+const crypto = require('crypto')
+const Secp256k1 = require('@lionello/secp256k1-js')
+
+import { KeyPair } from 'types'
+
+export function generateKeypair(): KeyPair {
+  let publicKey = null
+  let privateKey = null
+
+  do {
+    const privateKeyBuf = crypto.randomBytes(32)
+    privateKey = Secp256k1.uint256(privateKeyBuf, 16)
```

Comment:
Any chance this cuts off leading zeros to cause the length problem?

Reply:
I refactored this file, we're now using [paulmillr/noble-secp256k1](https://github.com/paulmillr/noble-secp256k1) instead of [lionello/secp256k1-js](https://github.com/lionello/secp256k1-js) which seems to be simpler to use and seems a little "established". Please test again when you have the time.